### PR TITLE
SALTO-1715: added deploy function to filter

### PR DESCRIPTION
**Pull request by alonstern**
Added a deploy function to filter to allow deploying changes through filters for the simple adapters deployment 

---
_Release Notes_: 
None

---
_User Notifications_: 
None

### CODE REVIEW

**Comment by adamro** [comment]
I am a bit out of context here, but why do we want/need a `deploy` filter?

**Comment by alonstern** [comment]
> I am a bit out of context here, but why do we want/need a `deploy` filter?

We would want to cover the "complex" cases of deployment in the adapters and not in the simple adapters infrastructure. Doing it using filters is convenient for us because:
* Each of these complex cases usually needs a "deploy" hook, in which it calls the deploy endpoints, and an "onFetch" hook  in which he tells that the type is deployable so it will pass the infrastructure change validator (by adding annotations to the type)
* It's a convenient way to separate each of these complex cases to a separate unit

File: packages/adapter-utils/src/filter.ts [hunk 1]
```diff
@@ -77,6 +84,27 @@ export const filtersRunner = <
     preDeploy: async changes => {
       await promises.array.series(filtersWith('preDeploy').reverse().map(filter => () => filter.preDeploy(changes)))
     },
+    /**
+     * deploy method for implementing a deployment functionality.
+     */
+    deploy: async changes => (
+      awu(filtersWith('deploy')).reduce(
+        async (total, current) => {
+          const res = await current.deploy(total.leftoverChanges)
```
Comment (noamhammer) [inline packages/adapter-utils/src/filter.ts:93]: super nit:
maybe `const { deployResult, leftoverChanges } = await current.deploy`

**Review by noamhammer — APPROVED** [review]
LGTM

**Pull request merged**